packages/server-wallet/src/models/app-bytecode.ts

```diff
@@ -18,6 +18,25 @@ export class AppBytecode extends Model implements RequiredColumns {
     return ['chain_id', 'app_definition'];
   }
 
+  static async upsertBytecode(
+    chainId: Bytes32,
+    appDefinition: Address,
+    appBytecode: Bytes32,
+    txOrKnex: TransactionOrKnex
+  ): Promise<AppBytecode> {
+    const insert = (
+      await txOrKnex(this.tableName).insert({chainId, appDefinition, appBytecode})
+    ).toString();
+    const update = txOrKnex(this.tableName)
+      .update({appBytecode})
+      .where({chainId, appDefinition});
+
+    return txOrKnex.raw(`? ON CONFLICT (chain_id,app_definition) DO ? returning *`, [
```

Comment:
Is it worth having a unit test verify that the on conflict clause works as intended?

Reply:
Good point! Done here: 334155f